### ISSUE TITLE
Write new test suite that does full testing.

### ISSUE THREAD

**Issue by alloy** [orta/cocoapods-keys#59]
When running the tests, build a version of the resulting `Keys.{h,m}` implementation into a CLI tool and use that during the tests to verify everything works as expected.

**Comment by alloy** [comment]
If you want to go full awesome, you could make it a C Ruby ext instead of a CLI tool.

**Comment by orta** [comment]
sounds like a nightmare to me :p

**Comment by alloy** [comment]
It’s actually not that hard, if you know someone that knows how that stuff works ;)

![](http://www.angel-art-and-gifts.com/images/heavens-angel-sleeping-puppy.jpg)

**Comment by davidbarsky** [comment]
If there’s any interest, I can write tests to replace the ones I removed in #68.

(Commenting to avoid duplicate work.)

**Comment by alloy** [comment]
@davidbarsky Hey, awesome. I was mostly looking for better coverage of the key master unit tests by using the tool I added in #61, see https://github.com/orta/cocoapods-keys/pull/61#issuecomment-90726313. Does that make sense to you?

**Comment by davidbarsky** [comment]
@alloy yeah, I think so. I’m thinking of attempting to generate empty keyrings, but fail upon trying to build them. Is that what you had in mind?

**Comment by alloy** [comment]
@davidbarsky I didn’t have much in mind with regards to the implementation, just from a high-level perspective that I wanted a simpler API that can be used in the unit tests to easily setup test objects.

**Comment by davidbarsky** [comment]
@alloy OK, I see what you’re getting at. alright, I’ll get to it.
